spec.html

```diff
@@ -22546,21 +22571,95 @@ <h1>ExecuteModule ( ) Concrete Method</h1>
+      </emu-clause>
+
+      <emu-clause id="sec-hostimportmoduledynamically" aoid="HostImportModuleDynamically">
+        <h1>Runtime Semantics: HostImportModuleDynamically ( _referencingScriptOrModule_, _specifier_, _promiseCapability_ )</h1>
+
+        <p>HostImportModuleDynamically is an implementation defined abstract operation that performs any necessary setup work in order to make available the module corresponding to the |ModuleSpecifier| String, _specifier_, occurring within the context of the script or module represented by the Script Record or Module Record _referencingScriptOrModule_. (_referencingScriptOrModule_ may also be *null*, if there is no active script or module when the `import()` expression occurs.) It then performs FinishDynamicImport to finish the dynamic import process.</p>
+
+        <p>The implementation of HostImportModuleDynamically must conform to the following requirements:</p>
+
+        <ul>
+          <li>
+            The abstract operation must always complete normally with *undefined*. Success or failure must instead be signaled as discussed below.
+          </li>
+          <li>
+            The host environment must conform to one of the two following sets of requirements:
+            <dl>
+              <dt>Success path</dt>
+
+              <dd>
+                <ul>
+                  <li>At some future time, the host environment must perform FinishDynamicImport(_referencingScriptOrModule_, _specifier_, _promiseCapability_, NormalCompletion(*undefined*)).</li>
+
+                  <li>Any subsequent call to HostResolveImportedModule after FinishDynamicImport has completed, given the arguments _referencingScriptOrModule_ and _specifier_, must complete normally.</li>
+
+                  <li>The completion value of any subsequent call to HostResolveImportedModule after FinishDynamicImport has completed, given the arguments _referencingScriptOrModule_ and _specifier_, must be a module which has already been evaluated, i.e. whose Evaluate concrete method has already been called and returned a normal completion.</li>
+                </ul>
+              </dd>
+
+              <dt>Failure path</dt>
+
+              <dd>
+                <ul>
+                  <li>At some future time, the host environment must perform FinishDynamicImport(_referencingScriptOrModule_, _specifier_, _promiseCapability_, an abrupt completion), with the abrupt completion representing the cause of failure.</li>
+                </ul>
+              </dd>
+            </dl>
+
+            <p>The intent of this specification is to not violate run to completion semantics. The spec-level formalization of this is a work-in-progress.</p>
```

Comment:
Is this work in progress something that should be completed before merging this PR, or do folks think it can land later?

Reply:
I think the invariants inside of the HostImportModuleDynamically and FinishDynamicImport do a good job of ensuring run-to-completion semantics. I don't think we need to block on further text. Probably this paragraph should be deleted during the merge--good catch.